src/backend/gporca/libgpopt/src/base/CColRefSet.cpp

```diff
@@ -126,6 +126,31 @@ CColRefSet::PcrFirst() const
 	return NULL;
 }
 
+//---------------------------------------------------------------------------
+//	@function:
+//		CColRefSet::PcrFirstUsed
+//
+//	@doc:
+//		Return first used member
+//
+//---------------------------------------------------------------------------
+CColRef *
+CColRefSet::PcrFirstUsed() const
+{
+	CColRefSetIter crsi(*this);
+	while (crsi.Advance())
+	{
+		CColRef *pcr = crsi.Pcr();
+
+		if (CColRef::EUsed == pcr->GetUsage())
+		{
+			return pcr;
+		}
+	}
+
+	return PcrFirst();
```

Comment:
As I understand the current solution  instead of selecting the first column of table (and marking it as used)  returns the first used column of it.
If I'm right, the next question is here: for ex. next case (seems it's impossible) - there are no selected columns in table (no one of columns marked as used), this function would return unmarked column and the problem still here. Is it so?

Reply:
> As I understand the current solution instead of selecting the first column of table (and marking it as used) returns the first used column of it.

Exactly

Reply:
> there are no selected columns in table (no one of columns marked as used), this function would return unmarked column and the problem still here. Is it so?

May be. In this case, I can mark the first column as used.

Reply:
fixed